packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -1931,6 +1939,50 @@ void main() {
     expect(getBorder(tester), disabledBorder);
   });
 
+  testWidgets('OutlineInputBorder radius carries over when lerping', (WidgetTester tester) async {
+    const Key key = Key('textField');
+
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: Material(
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: TextField(
+              key: key,
+              decoration: InputDecoration(
+                fillColor: Colors.white,
+                filled: true,
+                border: UnderlineInputBorder(
+                  borderSide: BorderSide(color: Colors.blue, width: 2.0),
+                  borderRadius: BorderRadius.zero,
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // TextField has the given border and doesn't ever tween to the default
+    expect(getBorderRadius(tester), BorderRadius.zero);
+    await tester.pump(const Duration(milliseconds: 50));
+    expect(getBorderRadius(tester), BorderRadius.zero);
+    await tester.pump(const Duration(milliseconds: 50));
+    expect(getBorderRadius(tester), BorderRadius.zero);
+    await tester.pumpAndSettle();
+    expect(getBorderRadius(tester), BorderRadius.zero);
+
+    // Focusing does not change the border either
+    await tester.tap(find.byKey(key));
+    expect(getBorderRadius(tester), BorderRadius.zero);
+    await tester.pump(const Duration(milliseconds: 50));
+    expect(getBorderRadius(tester), BorderRadius.zero);
+    await tester.pump(const Duration(milliseconds: 50));
+    expect(getBorderRadius(tester), BorderRadius.zero);
```

Comment:
This line is where it fails if I remove my fix.  If I change the first pump to 100 and get rid of this one, it doesn't fail at all!  It's a really weird timing thing.  Let me know if there's a better way to test it.

Reply:
Pump one frame to "get the animation started" and then you can pump frames with arbitrary durations.